src/util.ts

```diff
@@ -0,0 +1,327 @@
+  if (log.match(/eth_getBlockByNumber/i)) return true;
+  if (log.match(/eth_getBalance/i)) return true;
+  if (log.match(/processing height/i)) return true;
+  if (log.match(/new last processed height/i)) return true;
+  if (log.match(/eth_unsubscribe/i)) return true;
+  if (log.match(/eth_subscribe/i)) return true;
+  if (log.match(/new blocks subscription is quiet, rebuilding/i)) return true;
+  if (log.match(/received new chain header/i)) return true;
+  if (log.match(/dropping new height/i)) return true;
+
+  return false;
+};
+
+export interface PipeOptions {
+  message?: string;
+  fails?: {
+    message: string;
+    hint: string;
+  };
+  verbose?: boolean;
+  silent?: boolean;
+  emitter?: EventEmitter;
+  readyEvent?: string;
+}
+
+export const pipeNamedSubprocess = async function (
+  prefix: string,
+  prcss: ChildProcess,
+  options?: PipeOptions
+) {
+  let ready = !(options && options.message);
+  const fails = options?.fails;
+  const verbose = options?.verbose;
+  const silent = options?.silent;
+
+  if (!(prcss.stdout instanceof Readable && prcss.stderr instanceof Readable)) {
+    throw new Error("cannot pipe subprocess with out stdout and stderr");
+  }
+
+  prcss.stdout.on("data", function (data: string) {
+    // data is going to be a buffer at runtime
+    data = data.toString();
+    if (!data) return;
+
+    let lines = data.split("\n");
+    if (!verbose) {
+      lines = lines.filter((line) => !isExtraneousLog(line));
+      // if not verbose we are going to elliminate multiple empty
+      // lines and any messages that don't have at least one character
+      if (!lines.filter((line) => line.trim()).length) {
+        lines = [];
+      } else {
+        lines = lines.reduce((acc, cur) => {
+          if (acc.length && !acc[acc.length - 1] && !cur.trim()) return acc;
+
+          // @ts-ignore
+          return acc.concat([cur.trim()]);
+        }, []);
+      }
+    }
+
+    if (!silent) {
+      for (let i = 0; i < lines.length; i++) {
+        const line = lines[i];
+        if (!verbose && isExtraneousLog(line)) continue;
+        console.log(`[${prefix}] ${line}`);
+      }
+    }
+
+    if (!ready) {
+      if (
+        options &&
+        typeof options.message === "string" &&
+        data.includes(options.message) &&
+        typeof options.readyEvent === "string" &&
+        options.emitter instanceof EventEmitter
+      ) {
+        options.emitter.emit(options.readyEvent);
+        ready = true;
+      }
+    }
+  });
+
+  prcss.stderr.on("data", function (data: string) {
+    if (!(data && data.toString)) return;
+    // data is going to be a buffer at runtime
+    data = data.toString();
+    if (!data.trim()) return;
+
+    const lines = data
+      .split("\n")
+      .map((l) => l.trim())
+      .filter((l) => l);
+    for (let i = 0; i < lines.length; i++) {
+      const line = lines[i];
+      console.error(`[${prefix}] ${line}`);
+    }
+    if (fails && data.includes(fails.message)) {
```

Comment:
This enables us to specify that some messages mean that the network will never start correctly.  Basically this is checking for a message that indicates Docker isn't available.  Without something like this the process starting the local net will just hang forever instead of giving a nice error message indicating the problem